tests/test_mlflow_integration.py

```
@@ -0,0 +1,742 @@
+    assert config.log_artifacts is True
+    assert config.log_attribution is True
+
+
+def test_ensure_mlflow_available_reports_optional_dependency_guidance(
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    original_import = builtins.__import__
+
+    def _blocked_import(
+        name: str,
+        globals: dict[str, object] | None = None,
```

Ruff (0.16.1)

[error] Function argument `globals` is shadowing a Python builtin

(A002)